lib/slim-rails.rb

```diff
@@ -8,10 +8,14 @@ class Railtie < ::Rails::Railtie
       config.app_generators.template_engine :slim
 
       initializer 'slim_rails.configure_template_digestor' do |app|
-        if config.respond_to?(:assets)
-          config.assets.configure do |env|
-            env.assets.register_engine '.slim', Slim::Template
+        if ::Rails::VERSION::MAJOR == 3
+          if app.assets && app.assets.respond_to?(:register_engine)
+            app.assets.register_engine('.slim', Slim::Template)
           end
+        else
+          config.assets.configure do |env|
+            env.register_engine('.slim', Slim::Template)
+          end if config.respond_to?(:assets)
         end
```

Comment:
Exactly, it only makes sense in Assets Pipeline. So we  need a 'special case' for Rails 3.1 and Rails 3.2 or only for one of those?